portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -30,32 +30,32 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-
-
-
 /** Servlet that returns thread of comments. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
-  //private static final List<Comments> commentList = new ArrayList<>();
   private static final Gson gson = new Gson();
 
-
   @Override 
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Query query = new Query("Comments").addSort("timestamp", SortDirection.DESCENDING);
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    List<Comments> commentList = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
-      long id = entity.getKey().getId();
-      String comment = (String) entity.getProperty("comment");
-      long timestamp = (long) entity.getProperty("timestamp");
-
-      Comments comments = new Comments(id, comment, timestamp);
-      commentList.add(comments);
+    int maxCommentNum = getMaxComments(request);
+    if(maxCommentNum == -1){
+        System.out.println("could not convert to int");
     }
 
+    final List<Comments> commentList = new ArrayList<Comments>();
+    for (Entity entity : results.asIterable()) {
+        if(commentList.size() < maxCommentNum){
```

Comment:
This works, but it loops through the entire list of results every time. More efficient if you just do
for(Entity entity: results...) {
  if(commentList.size() >= maxCommentNum) {
    break;
  }
  ...
}

The `break` will exit the for loop